src/components/Button.tsx

```diff
@@ -79,9 +87,11 @@ export function Button(props: ButtonProps) {
 
   const buttonContent = (
     <>
-      {icon && <Icon xss={iconStyles[size]} icon={icon} />}
-      {label}
-      {endAdornment && <span css={Css.ml1.$}>{endAdornment}</span>}
+      {icon && <Icon xss={iconStyles[size]} icon={inFlightIcon && asyncInProgress ? inFlightIcon : icon} />}
+      {inFlightLabel && asyncInProgress ? inFlightLabel : label}
+      {endAdornment && (
+        <span css={Css.ml1.$}>{inFlightEndAdornment && asyncInProgress ? inFlightEndAdornment : endAdornment}</span>
+      )}
```

Comment:
**thought:** This is where I think that maybe we _always_ do the `<Loader />` end adornment if `asyncInProgress`. So it'd be more like:
```suggestion
      {(endAdornment || asyncInProgress) && (
        <span css={Css.ml1.$}>{asyncInProgress ? inFlightEndAdornment : endAdornment}</span>
      )}
```